src/server/ApiMessage.cpp

```diff
@@ -145,10 +152,17 @@ try
         ULONG cbWriteSize = Descriptor.OutputSize - State.WriteOffset;
         RETURN_IF_FAILED(ULongMult(cbWriteSize, cbFactor, &cbWriteSize));
 
+        // If we were previously called with a huge buffer we have an equally large _outputBuffer.
+        // We shouldn't just keep this huge size around, if no one needs it anymore.
+        if (_outputBuffer.capacity() > 16 * 1024 && (_outputBuffer.capacity() >> 1) > cbWriteSize)
+        {
+            _outputBuffer.shrink_to_fit();
+        }
+
         _outputBuffer.resize(cbWriteSize);
 
         // 0 it out.
-        std::fill(_outputBuffer.begin(), _outputBuffer.end(), (BYTE)0);
+        std::fill_n(_outputBuffer.data(), _outputBuffer.size(), BYTE(0));

```

Comment:
This was sort of a programming error that I fixed on the side.
`<algorithm>` doesn't work properly with non-STL containers, so it's important that we pass it raw pointers if we can. Otherwise it can't figure out what to do with it and sets each byte to 0 one by one.
This one will now properly compile to a single call to `memset()`.

Reply:
Excellent. Thanks!